src/main/java/pl/javastart/task/logic/University.java

```diff
@@ -0,0 +1,150 @@
+package pl.javastart.task.logic;
+
+import pl.javastart.task.models.Grade;
+import pl.javastart.task.models.Group;
+import pl.javastart.task.models.Lecturer;
+import pl.javastart.task.models.Student;
+
+public class University {
+    private static final int MAX_STUDENTS_AMOUNT_IN_GROUP = 15;
+    private static final int MAX_CLASSES_AMOUNT = 15;
+    private Student[] students;
+    private Lecturer[] lecturers;
+    private Group[] groups;
+
+    public University(int maxStudentsAmount, int maxLecturersAmount, int maxGroupsAmount) {
+        this.students = new Student[maxStudentsAmount];
+        this.lecturers = new Lecturer[maxLecturersAmount];
+        this.groups = new Group[maxGroupsAmount];
+    }
+
+    public Student addStudent(String firstName, String lastName, int index) {
+        Student student = new Student(firstName, lastName, index, MAX_CLASSES_AMOUNT);
+        for (int i = 0; i < students.length; i++) {
+            if (students[i] == null) {
+                students[i] = student;
+                return student;
+            }
+        }
```

Comment:
Zamiast przy wstawianiu każdego studenta na nowo szukać miejsca, w które go wstawić, lepiej dodaj to tej klasy pola typu `studentsCount`, które będą inkrementowane wraz z dodawaniem kolejnych studentów, czy prowadzących.
Będzie wtedy tutaj po prostu `students[studentsCount] = new Student(...)`